parse.go

```diff
@@ -119,6 +120,51 @@ func flags() []string {
 	return os.Args[1:]
 }
 
+// DefaultEnvNameFunc is a function that takes a field and returns the default name for the environment variable that sets its value.
```

Comment:
For now, no need to export any of the new types or functions in this file. Could the new config field just have the type `func(field reflect.StructField) string` directly inlined within the struct?

Reply:
Yup, no problem, I'm generally a fan of DRY but if you prefer to repeat it explicitly that's fine